### DISCUSSION TITLE
Including API titles in the browser-compat-data

### DISCUSSION THREAD

**Discussion by kzar** [mdn/mdn-community#669] [Browser compatibility data] [answered]
For a recent project, I had a list of APIs (e.g. `api.Navigator.clearOriginJoinedAdInterestGroups`) as named in browser-compat-data and I needed to know the parent API title (e.g. `Protected Audience API`). In the end, I scraped the linked MDN docs and specifications to create an API title lookup for this.

- Perhaps having API titles in browser-compat-data would make sense? It seems very useful to me.
- If so, I have [attached the apititles.json API title lookup](https://github.com/mdn/mdn-community/files/14959090/apititles.json) I made for use as a starting point.

_(I went through the lookup and added missing API titles + tried to correct the wrong entries. But I didn't have a ton of time to do this, so there likely are some mistakes in there and some API titles that are still missing.)_

**Comment by teoli2003** [comment, marked as answer]
There is a process to add such "groups" into BCD. I think this may solve your use case.

The WebDX group will define the tags (one per API), as part of their [web-features](https://github.com/web-platform-dx/web-features) project and they will be added to mdn/browser-compat-data. For example, this is how they define the [Fullscreen API](https://github.com/web-platform-dx/web-features/blob/main/feature-group-definitions/fullscreen.yml).

This is still a WIP, but regular work is being done for it.

![Capture d’écran 2024-04-12 à 15 45 37](https://github.com/mdn/mdn-community/assets/1466293/e62f3e50-0701-4549-8c1c-24efdda2c1d6)

**Reply by kzar** [reply]
Ah right, cool OK. Would the lookup I attached help them with that work?

**Reply by teoli2003** [reply]
Lets ask @Elchi3 or @ddbeck.

**Reply by ddbeck** [reply]
Yes, this is almost precisely what we're working on with [web-features](https://github.com/web-platform-dx/web-features/).

Right now, we publish a set of higher-level "features" (higher level relative to a lone BCD key, usually). So if you have the key `api.AbortController.AbortController` (for the `AbortController` constructor), then you can use the tag (`web-features:aborting`) to find that it's part of a broader feature made up `AbortController` and `AbortSignal`.

But we've also started work on "groups" and "snapshots", to capture ideas like "this compat key is part of a broader group of abortable things" or "this method is part of the 2015 JavaScript spec" or what-have-you. That work is still immature (I'm focused presently on defining features that have become interopable on the web platform in the last couple of years), but I hope to turn more attention to it soon. Of course, if you have an application for the groups and snapshots, that'd be interesting to know about and can help us prioritize for the web-features data consumers.

**Reply by kzar** [reply]
Cool OK, got it. So the project I was working on was to do with comparing the results of [mdn-bcd-collector](https://mdn-bcd-collector.gooborg.com/) between browsers, to see the differences in available APIs. To help make sense of those differences (sometimes hundreds), I needed to know which underlying feature a group of APIs related to. That's why I ended up making the attached [apititles.json](https://github.com/mdn/mdn-community/files/14959090/apititles.json) lookup. So yea, if that lookup can help with your work on web-features please feel free to use it!

**Comment by pepelsbey** [comment]
Thank you for starting the discussion. I’m closing this now due to inactivity and one of the comments marked as an answer, but if you think that’s a mistake, feel free to reply. You’re also welcome to join the [MDN Discord](https://developer.mozilla.org/discord) if you’d prefer to chat with us more about it in sync.
